source/solvers/cahn_hilliard.cc

```diff
@@ -1185,6 +1230,81 @@ CahnHilliard<dim>::calculate_barycenter(const GlobalVectorType &solution,
                                                    barycenter_velocity);
 }
 
+template <int dim>
+void
+CahnHilliard<dim>::apply_phase_filter()
+{
+  auto mpi_communicator = this->triangulation->get_communicator();
+
+  FEValues<dim> fe_values(*mapping,
+                          *fe,
+                          *cell_quadrature,
+                          update_values | update_gradients |
+                            update_quadrature_points | update_JxW_values);
+
+  const FEValuesExtractors::Scalar phase_order(0);
+
+  TrilinosWrappers::MPI::Vector filtered_solution_owned(
+    this->locally_owned_dofs, mpi_communicator);
+  filtered_solution_owned = this->present_solution;
+
+  filtered_solution.reinit(this->present_solution);
+
+  std::unordered_map<unsigned int, bool> filtered_cell_list;
```

Comment:
Instead of yusing and std::unordered_map of bool use an unordered_set (which is exactly that in a way...)
If it's in the set it's true, if it's not it is false.

Reply:
agreed